backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/user/api/impl/UserDAO.java

```diff
@@ -318,4 +326,75 @@ public void save(@NonNull final org.compiere.model.I_AD_User user)
 		return OrgMappingId.optionalOfRepoId(sourceUserRecord.getAD_Org_Mapping_ID());
 	}
 
+	@Override
+	public void deleteUserPreferenceByUserId(final UserId userId)
+	{
+		queryBL.createQueryBuilder(I_AD_Preference.class)
+				.addEqualsFilter(I_AD_Preference.COLUMNNAME_AD_User_ID, userId)
+				.create()
+				.delete();
+	}
+
+	@Override
+	public void deleteUserOrgAccessByUserId(final UserId userId)
+	{
+		queryBL.createQueryBuilder(I_AD_User_OrgAccess.class)
+				.addEqualsFilter(I_AD_User_OrgAccess.COLUMNNAME_AD_User_ID, userId)
+				.create()
+				.delete();
+	}
+
+	@Override
+	public void deleteUserSubstituteByUserId(final UserId userId)
+	{
+		queryBL.createQueryBuilder(I_AD_User_Substitute.class)
+				.addEqualsFilter(I_AD_User_Substitute.COLUMNNAME_AD_User_ID, userId)
+				.create()
+				.delete();
+	}
+
+	@Override
+	public void deleteUserMailByUserId(final UserId userId)
+	{
+		queryBL.createQueryBuilder(I_AD_UserMail.class)
+				.addEqualsFilter(I_AD_UserMail.COLUMNNAME_AD_User_ID, userId)
+				.create()
+				.delete();
+	}
+
+	@Override
+	public void deleteUserQueryByUserId(final UserId userId)
+	{
+		queryBL.createQueryBuilder(I_AD_UserQuery.class)
+				.addEqualsFilter(I_AD_UserQuery.COLUMNNAME_AD_User_ID, userId)
+				.create()
+				.delete();
+	}
+
+	@Override
+	public void deleteUserRolesByUserId(final UserId userId)
+	{
+		queryBL.createQueryBuilder(I_AD_User_Roles.class)
+				.addEqualsFilter(I_AD_User_Roles.COLUMNNAME_AD_User_ID, userId)
+				.create()
+				.delete();
+	}
+
+	@Override
+	public void deleteUserOrgAssignmentByUserId(final UserId userId)
+	{
+		queryBL.createQueryBuilder(I_C_OrgAssignment.class)
+				.addEqualsFilter(I_C_OrgAssignment.COLUMNNAME_AD_User_ID, userId)
+				.create()
+				.delete();
+	}
+
+	@Override
+	public void deletePInstancesForUserId(final UserId userId)
+	{
+		queryBL.createQueryBuilder(I_AD_PInstance.class)
```

Comment:
change the AD_PInstance.AD_User_ID:
* drop the FK
* in Table & Columns, for AD_user_ID column, flag it as no FK

Reply:
FK droped